### ISSUE TITLE
NullPointerException During Scan

### ISSUE THREAD

**Issue by asleson** [lukehutch/fast-classpath-scanner#91]
Hello,

I just started using FastClaspathScanner v2.0.6.  It's working great so far, but in one particular case it's always throwing a `NullPointerException`.  I'm using the `SubinterfaceMatchProcessor`:

```
final FastClasspathScanner scanner = new FastClasspathScanner(getRootPackage());
scanner.matchSubinterfacesOf(JpaRepository.class, new SubinterfaceMatchProcessor<JpaRepository>() {

  @Override
  public void processMatch(Class<? extends JpaRepository> type)
  {
    repositories.add(type);
  }
}).scan();
```

I've verified that the `getRootPackage()` method is returning a valid package name.

This is the relevant part of the stack trace:

```
Caused by: java.lang.RuntimeException: java.lang.NullPointerException
	at io.github.lukehutch.fastclasspathscanner.FastClasspathScanner.scan(FastClasspathScanner.java:1033)
	at io.github.lukehutch.fastclasspathscanner.FastClasspathScanner.scan(FastClasspathScanner.java:1061)
	at io.github.lukehutch.fastclasspathscanner.FastClasspathScanner.scan(FastClasspathScanner.java:1084)
	at com.biperf.cbr.component.ScanningCbrComponent.getJpaRepositoryClasses(ScanningCbrComponent.java:56)
	at com.biperf.cbr.component.ScanningCbrComponent.getRespositoryBindings(ScanningCbrComponent.java:65)
	at com.biperf.cbr.runtime.guice.OverrideAwareModule.configure(OverrideAwareModule.java:53)
	at com.google.inject.AbstractModule.configure(AbstractModule.java:62)
	at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:253)
	at com.google.inject.spi.Elements.getElements(Elements.java:108)
	at com.google.inject.internal.InjectorShell$Builder.build(InjectorShell.java:135)
	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:104)
	... 57 more
Caused by: java.lang.NullPointerException
	at io.github.lukehutch.fastclasspathscanner.scanner.ScanSpec$6.lookForMatches(ScanSpec.java:916)
	at io.github.lukehutch.fastclasspathscanner.scanner.ScanSpec.callMatchProcessors(ScanSpec.java:432)
	at io.github.lukehutch.fastclasspathscanner.scanner.Scanner.call(Scanner.java:365)
	at io.github.lukehutch.fastclasspathscanner.scanner.Scanner.call(Scanner.java:59)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	... 1 more

```

If it helps at all, in this particular case, I'm not expecting the scan to produce any results.  I use the `SubinterfaceMatchProcessor` in another area that does produce results, and there's no problem.

Is there something I'm doing wrong?  I can't figure it out.

Thank you!!

**Comment by lukehutch** [comment]
Hi @asleson, thanks for this bug report. Nope, you're not doing anything wrong, that's a genuine bug :-)  Fortunately it was a one-line fix (I missed one change in the API when I switched from returning null to returning the empty list at one point, months ago). Can you please test with the latest git master version? Thanks!

**Comment by asleson** [comment]
Yes, that appears to have fixed the problem.  **Thank you!!**  Will there soon be a bug fix release, or are you waiting on other work?  I don't want to be a nag, I just want to know if I need to tweak my build process now or just wait it out.

**Comment by lukehutch** [comment]
No problem. I'll push out another release as soon as #90 is solved, but I don't have any experience with JBoss, so I'm not the best one to write that patch. If you have experience with JBoss and would like to help the reporter of that bug fix it, your help would be greatly appreciated! Either way, thanks for this bug report!

**Comment by asleson** [comment]
I don't know anything about classpath scanning, but maybe look at the [ResolverUtil](https://github.com/StripesFramework/stripes/blob/master/stripes/src/main/java/net/sourceforge/stripes/util/ResolverUtil.java) class in the [Stripes](https://github.com/StripesFramework/stripes) framework.  I started using `FastClasspathScanner` as a replacement for `ResolverUtil` because I didn't want to have a dependency on a web framework just for the scanner.  I see that `ResolverUtil` has dependencies on classes in the [vfs](https://github.com/StripesFramework/stripes/tree/master/stripes/src/main/java/net/sourceforge/stripes/vfs) package, which just happens to mention JBoss.

I don't use JBoss, but I remember from years past that the Stripes team had to do some work to get ResolverUtil working correctly on JBoss.

**Comment by lukehutch** [comment]
Thanks, that's a useful pointer. I'll try to merge their approach with the one that's already in place.

Keep an eye out for the next release, hopefully soon. Thanks!

**Comment by lukehutch** [comment]
@asleson Released in 2.0.7. Thanks again!
